### PR TITLE
PWN-6078 - Add send screen stub

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6078

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/send/ui/NoTokenSendFragment.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+private const val EXTRA_TOKEN = "EXTRA_TOKEN"
+
+class NoTokenSendFragment : BaseFragment(R.layout.fragment_send_no_token) {
+
+    companion object {
+        fun create(token: Token) = NoTokenSendFragment().withArgs(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/NoTokenSendFragment.kt:22]: explicit return type

File: app/src/main/java/org/p2p/wallet/send/ui/NoTokenSendFragment.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import org.p2p.wallet.home.model.Token
+import org.p2p.wallet.moonpay.ui.new.NewBuyFragment
+import org.p2p.wallet.receive.solana.ReceiveSolanaFragment
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+private const val EXTRA_TOKEN = "EXTRA_TOKEN"
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/NoTokenSendFragment.kt:17]: EXTRA_DEFAULT_TOKEN

File: app/src/main/java/org/p2p/wallet/send/ui/NoTokenSendFragment.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+class NoTokenSendFragment : BaseFragment(R.layout.fragment_send_no_token) {
+
+    companion object {
+        fun create(token: Token) = NoTokenSendFragment().withArgs(
+            EXTRA_TOKEN to token
+        )
+    }
+
+    private val binding: FragmentSendNoTokenBinding by viewBinding()
+    private val token: Token by args(EXTRA_TOKEN)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/NoTokenSendFragment.kt:28]: token -> defaultTokenToBuy

File: app/src/main/java/org/p2p/wallet/send/ui/NoTokenSendFragment.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            toolbar.setNavigationOnClickListener {
+                popBackStack()
+            }
+            buttonBuy.setOnClickListener {
+                replaceFragment(NewBuyFragment.create(token))
+            }
+
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/NoTokenSendFragment.kt:40]:
```suggestion
```

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 5]
```diff
@@ -132,6 +134,16 @@ class HomePresenter(
         }
     }
 
+    override fun onSendClicked() {
+        fallbackUsdcTokenForBuy?.let { token ->
+            if (state.tokens.isEmpty()) {
+                view?.showSendNoToken(token)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:140]: showSendNoToken -> showSendNoTokens

File: app/src/main/java/org/p2p/wallet/send/ui/NoTokenSendFragment.kt [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import org.p2p.wallet.receive.solana.ReceiveSolanaFragment
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+private const val EXTRA_TOKEN = "EXTRA_TOKEN"
+
+class NoTokenSendFragment : BaseFragment(R.layout.fragment_send_no_token) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/NoTokenSendFragment.kt:19]: NoTokens - because user has no tokens, not some particular token

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 7]
```diff
@@ -132,6 +134,16 @@ class HomePresenter(
...
     }
 
+    override fun onSendClicked() {
+        fallbackUsdcTokenForBuy?.let { token ->
+            if (state.tokens.isEmpty()) {
+                view?.showSendNoTokens(token)
+            } else {
+                view?.navigateToSend(newSendEnabledFeatureToggle.isFeatureEnabled)
+            }
+        }
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:144]: This is wrong. 
First, you should check `newSendEnabledFeatureToggle.isFeatureEnabled`. 
If new send is **disabled**, then you just open old send fragment
if new send is **enabled**, then you check if you user has tokens or not.

What is `fallbackUsdcTokenForBuy`? I feel like you don't need it

File: app/src/main/java/org/p2p/wallet/send/ui/NoTokensSendFragment.kt [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import org.p2p.wallet.home.model.Token
+import org.p2p.wallet.moonpay.ui.new.NewBuyFragment
+import org.p2p.wallet.receive.solana.ReceiveSolanaFragment
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+private const val EXTRA_DEFAULT_TOKEN = "EXTRA_DEFAULT_TOKEN"
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/send/ui/NoTokensSendFragment.kt:18]:
```suggestion
private const val ARG_DEFAULT_TOKEN = "ARG_DEFAULT_TOKEN "
```

File: app/src/main/java/org/p2p/wallet/send/ui/NoTokensSendFragment.kt [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+class NoTokensSendFragment : BaseFragment(R.layout.fragment_send_no_token) {
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/send/ui/NoTokensSendFragment.kt:20]: Maybe we should start naming with `Send`? 
For example: `SendDisabledFragment` or smth like that. 
The developer can have a context starting from the first word in the name

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/send/ui/NoTokensSendFragment.kt (hunk 9 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/send/ui/NoTokensSendFragment.kt:20]: im for SendNoTokensFragment

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 10]
```diff
@@ -132,6 +134,19 @@ class HomePresenter(
...
     }
 
+    override fun onSendClicked() {
+        if (newSendEnabledFeatureToggle.isFeatureEnabled) {
+            if (state.tokens.isEmpty()) {
+                view?.showSendNoTokens(fallbackUsdcTokenForBuy ?: return)
+            } else {
+                view?.navigateToSend(isNewSendAvailable = true)
+            }
+            fallbackUsdcTokenForBuy
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:144]: remove?

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 11]
```diff
@@ -132,6 +134,19 @@ class HomePresenter(
         }
     }
 
+    override fun onSendClicked() {
+        if (newSendEnabledFeatureToggle.isFeatureEnabled) {
+            if (state.tokens.isEmpty()) {
+                view?.showSendNoTokens(fallbackUsdcTokenForBuy ?: return)
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:140]: In this case if it's null the user will be stuck on the screen? 
If user clicks the app will do anything?

**Pull request merged**